Firmware/Marlin_main.cpp

```diff
@@ -2905,7 +2906,7 @@ static void gcode_G80()
 
         // Move Z up to the probe height of the current Z point.
         const float z0 = mbl.z_values[iy][ix];
-        const float init_z_bckp = !has_z ? MESH_HOME_Z_SEARCH : z0 + 0.35;
+        const float init_z_bckp = !has_z ? MESH_HOME_Z_SEARCH : z0 + MESH_HOME_Z_SEARCH_FAST;
         if (init_z_bckp > current_position[Z_AXIS])
             current_position[Z_AXIS] = init_z_bckp;
         plan_buffer_line_curposXYZE(Z_LIFT_FEEDRATE);
```

Comment:
Can we add `plan_buffer_line_curposXYZE` call and `st_synchronize` under the if statement? It looks like to me this planning is not needed unless the if statement is true.

Reply:
You are correct. Pushing the change now 👍